src/commcare_cloud/ansible/roles/formplayer/tasks/main.yml

```
@@ -41,7 +41,7 @@
     # but leaving any that have been modified in the last 20m
     # (to avoid a race condition on newly created dirs)
     # Also never delete {{ formplayer_access_log_dir }}
-    job: 'find {{ formplayer_data_dir }} -path {{ formplayer_access_log_dir }} -prune -o -empty -type d -mmin +20 -delete'
+    job: 'find {{ formplayer_data_dir }} -not -path {{ formplayer_access_log_dir }} -empty -type d -mmin +20 -delete'
```

I think the `-path` option will not match in this new command. A wildcard may fix it though.
```suggestion
    job: "find {{ formplayer_data_dir }} -not -path '{{ formplayer_access_log_dir }}/*' -empty -type d -mmin +20 -delete"
```

---

No, adding the `*` will delete the access log directory if it is empty:

```bash
$ mkdir dontdelete delete
$ find . -not -path ./dontdelete -empty -type d
./delete

$ find . -not -path ./dontdelete/* -empty -type d
./delete
./dontdelete
```

The `-path` predicate tests for a literal end-to-end match, so its argument should be the exact path we do not want to delete. All else (below `{{ formplayer_data_dir }}`) that is an empty directory is deleted.

---

I just added a commit that tackles the not-so-corner case that `{{ formplayer_data_dir }}` itself is empty and would get deleted.

---

> All else below `formplayer_data_dir` that is an empty directory will be deleted.

I don't think that was the original intent. The documentation for `-prune` says

> if  the file is a directory, do not descend into it. ... For example,  to  skip  the  directory  `src/emacs` and all files and directories under it, and print the names of the other files found, do something like this:
>
>     find . -path ./src/emacs -prune -o -print

With the previous code, `formplayer_access_log_dir` and all subdirectories would have been ignored (except they were not because the `-delete` command implies `-depth`, which renders `-prune` ineffective). Some examples to illustrate:

```sh
## setup directories for testing
$ mkdir -p temp/data/logs/elk
$ mkdir -p temp/data/deer

## -prune example : does not descend into "logs" dir
$ find temp/data -path temp/data/logs -prune -o -empty -type d -print
temp/data/deer

## -not -path example : descends into "logs" dir
$ find temp/data -mindepth 1 -not -path temp/data/logs -empty -type d -print
temp/data/deer
temp/data/logs/elk
```

However, you are correct that simply adding `/*` to the path will cause an empty logs directory to be deleted. To avoid that we need another `-not -path ...` predicate. Adding a suggestion in a separate comment...